velox/exec/Spiller.h

```diff
@@ -70,27 +110,32 @@ class Spiller {
             pool,
             spillMappedMemory()),
         pool_(pool),
-        executor_(executor) {}
+        executor_(executor) {
+    // TODO: add to support kHashJoin type later.
+    VELOX_CHECK_NE(
+        type_,
+        Type::kHashJoin,
+        "Spiller type:{} is not supported yet",
+        TypeName(type_));
+    // kOrderBy spiller type must only have one partition.
+    VELOX_CHECK((type_ != Type::kOrderBy) || (state_.maxPartitions() == 1));
+    spillRuns_.reserve(state_.maxPartitions());
+    for (int i = 0; i < state_.maxPartitions(); ++i) {
+      spillRuns_.emplace_back(spillMappedMemory());
+    }
+  }
 
   // Spills rows from 'this' until there are under 'targetRows' rows
   // and 'targetBytes' of allocated variable length space in
-  // use. 'iterator' should be at the start of 'container_' on first
-  // call.  spill() starts with one spill partition and initializes
+  // use. spill() starts with one spill partition and initializes
   // more spill partitions as needed to hit the size target. If there
   // is no more data to spill in one hash partition, it starts spilling
   // another hash partition until all hash partitions are spilling. 'bits'
   // specifies the bit field of the hash number of a row that determines which
   // hash partition the row belongs to. A spillable hash partition has a
   // SpillRun struct in 'spillRuns_' A targetRows of 0 causes all data to be
   // spilled and 'container_' to become empty.
-  void spill(
-      uint64_t targetRows,
-      uint64_t targetBytes,
-      RowContainerIterator& iterator);
-
-  bool isSpilled(int32_t partition) const {
-    return state_.hasFiles(partition);
-  }
+  void spill(uint64_t targetRows, uint64_t targetBytes);
```

Comment:
Perhaps, update the comment to accommodate the case when there are no 'bits' and clarify the order in which partitions are selected.